src/Hintman/Suggestion/Core.hs

```diff
@@ -0,0 +1,22 @@
+module Hintman.Suggestion.Core
+       ( Line (..)
+       , LineState (..)
+       , Suggestion (..)
+       ) where
+
+
+data LineState = Plain | Trailing
+    deriving (Eq, Show)
+
+data Line = Line
+    { lineRaw   :: Int
+    , lineBody  :: Text
+    , lineState :: LineState
```

Comment:
I guess this is not needed in this data type. `Line` should contain only info about the line itself.